research/fed-ce/README.md

```diff
@@ -10,7 +10,54 @@ Accepted to [CVPR2023](https://cvpr2023.thecvf.com/).
 
 ## License
-- TBD
+- The code in this directory is released under Apache v2 License.
+
+## Multi-source Prostate Segmentation
+This example uses 2D (axial slices) segmentation of the prostate in T2-weighted MRIs based on multiple datasets.
+
+Please refer to [Prostate Example](https://github.com/NVIDIA/NVFlare/tree/dev/examples/advanced/prostate) for details of data preparation and task specs. In the following, we assume the data has been prepared in the same way to `${PWD}/data_preparation`. The dataset is saved to `${PWD}/data_preparation/dataset_2D`, and datalists are saved to `${PWD}/data_preparation/datalist_2D`.
+
+## Setup
+
+Install required packages for training
+```
+pip install --upgrade pip
+pip install -r ./requirements.txt
+```
+
+## Run automated experiments
+We use the NVFlare simulator to run FL training automatically, the 6 clients are named `client_I2CVB,client_MSD,client_NCI_ISBI_3T,client_NCI_ISBI_Dx,client_Promise12,client_PROSTATEx`.
+
+### Prepare local configs
+First, we add the image directory root to `config_train.json` files for generating the absolute path to dataset and datalist. 
+In the current folder structure, it will be `${PWD}/..`. 
+It can be any arbitrary path where the data locates.  
+```
+for job in fedce_prostate
+do
+  sed -i "s|DATASET_ROOT|${PWD}/data_preparation|g" jobs/${job}/app/config/config_train.json
+done
+```
+### Use NVFlare simulator to run the experiments
+We use NVFlare simulator to run the FL training experiments, following the pattern:
+```
+nvflare simulator jobs/[job] -w ${workspace_path}/[job] -c [clients] -gpu [gpu] -t [thread]
+```
+`[job]` is the experiment job that will be submitted for the FL training. 
+In this example, this is `fedce_prostate`.  
+The combination of `-c` and `-gpu`/`-t` controls the resource allocation. 
+
+## Results on six clients for FedCE
+In this example, we run six clients on 2 GPUs. The minimum GPU memory requirement is 10 GB per GPU. We put the workspace in `/tmp` folder
+```
+nvflare simulator jobs/fedce_prostate -w /tmp/nvflare/fedce_prostate -c client_I2CVB,client_MSD,client_NCI_ISBI_3T,client_NCI_ISBI_Dx,client_Promise12,client_PROSTATEx -gpu 0,1,0,1,0,1
+```
+
+### Metrics and FedCE contribution estimation curves
+In this example, each client computes their validation scores using their own
+validation set. We also record the contribution scores for each client.
+The TensorBoard curves of validation Dice and contribution scores for the 100 epochs (100 rounds, 1 local epoch per round) during training are shown below:
+![All training curve](./figs/all_training.png)
```

Comment:
Can you add some interpretation/conclusion to explain the results?

Reply:
added some interpretation